src/index.js

```diff
@@ -66,14 +66,15 @@ export default class VueI18n {
 
     const locale: Locale = options.locale || 'en-US'
     const fallbackLocale: Locale = options.fallbackLocale || 'en-US'
-    const messages: LocaleMessages = options.messages || {}
     const dateTimeFormats = options.dateTimeFormats || {}
     const numberFormats = options.numberFormats || {}
+    const prefix: Object = this.prefixedConstructor(options.prefix)
+    const messages: LocaleMessages = this.createPrefixedMessages(prefix, options.messages)
 
     this._vm = null
     this._formatter = options.formatter || defaultFormatter
     this._modifiers = options.modifiers || {}
-    this._missing = options.missing || null
+    this._missing = prefix.active ? this.createPrefixedMissingHandler(prefix, options.missing) : options.missing || null
```

Comment:
I think current implemenetation is unneedfully tightly coupled with core.
What I wanted to suggest in https://github.com/kazupon/vue-i18n/pull/732#discussion_r338855181 is having a static method (factory method) in VueI18n and adding NO extra fields/methods to VueI18n.